src/field/pokemon.ts

```diff
@@ -5300,6 +5300,15 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
     );
   }
 
+  messageIsImmune(quiet: boolean): boolean {
+    if(!quiet){
+      globalScene.queueMessage(
+        i18next.t("abilityTriggers:moveImmunity", {pokemonNameWithAffix: getPokemonNameWithAffix(this),})
+      );
+    }
+    return false;
+  }
+
```

Comment:
```suggestion
  queueImmuneMessage(quiet: boolean): void {
    if(!quiet) {
      globalScene.queueMessage(
        i18next.t("abilityTriggers:moveImmunity", {pokemonNameWithAffix: getPokemonNameWithAffix(this),})
      );
    }
  }
```
This method shouldn't be used as a shortcut for when you're returning false because that makes its usages less clear. Just return false after calling the method

I would also suggest that you add a parameter to this method for the status attempting to be applied, because the message should be different depending on whether the status being applied is the same status that the pokemon already has.

For example, if your pokemon is already paralyzed and you try to paralyze it, you'll get "{pokemon} is already paralyzed!". If you try to give it a different status then you get the standard immune message.

Reply:
Alright, my thought process for that was to make the samllest addiciton to the base code. Now i can see how it could make it harder to follow the logic of the function by making those changes.